### PR TITLE
Added a note about not fetch users for every request

### LINKED ISSUE #5534: Document how to refresh the user *not* on every request
See symfony/symfony#1338

I think a short section in the custom user provider section should be enough, with the warning that doing this with, for example, Doctrine carries the risks outlined in that issue.

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #5534.

@weaverryan not sure if this is what you had in mind when you created #5534.

### CODE REVIEW

File: security/custom_provider.rst [hunk 1]
```diff
@@ -174,6 +174,14 @@ Here's an example of how this might look::
         }
     }
 
+.. note::
+
+    If fetching the user is a complex or slow operation, you can avoid making it
+    for every request and rely instead on some cache provided by yourself.
```
Comment (OskarStark) [inline security/custom_provider.rst]:
```suggestion
    for every request and instead rely on some cache provided by yourself.
```

??

**Comment by wouterj** [comment]
Hmm, not sure if this is a usefull addition to the docs. I think most applications use a doctrine user provider and people not using it have probably written their own user provider (and thus implemented the `refreshUser()` method themselves). I would consider this the "group that is experienced enough to write this themselves".

**Comment by javiereguiluz** [comment]
So, should we close this as "won't merge" and the related issue as "won't fix" ?

**Comment by javiereguiluz** [comment]
Closing as won't fix then. Thanks!

**Pull request closed without merging**